pkg/ebpf/filters.go

```diff
@@ -37,3 +41,318 @@ type Filter struct {
 	BinaryFilter      *filters.BPFBinaryFilter
 	Follow            bool
 }
+
+func NewFilterScope() *FilterScope {
+	return &FilterScope{
+		ID:                0,
+		EventsToTrace:     map[events.ID]string{},
+		UIDFilter:         filters.NewBPFUInt32Filter(UIDFilterMap),
+		PIDFilter:         filters.NewBPFUInt32Filter(PIDFilterMap),
+		NewPidFilter:      filters.NewBoolFilter(),
+		MntNSFilter:       filters.NewBPFUIntFilter(MntNSFilterMap),
+		PidNSFilter:       filters.NewBPFUIntFilter(PidNSFilterMap),
+		UTSFilter:         filters.NewBPFStringFilter(UTSFilterMap),
+		CommFilter:        filters.NewBPFStringFilter(CommFilterMap),
+		ContFilter:        filters.NewBoolFilter(),
+		NewContFilter:     filters.NewBoolFilter(),
+		ContIDFilter:      filters.NewContainerFilter(CgroupIdFilterMap),
+		RetFilter:         filters.NewRetFilter(),
+		ArgFilter:         filters.NewArgFilter(),
+		ContextFilter:     filters.NewContextFilter(),
+		ProcessTreeFilter: filters.NewProcessTreeFilter(ProcessTreeFilterMap),
+		BinaryFilter:      filters.NewBPFBinaryFilter(BinaryFilterMap),
+		Follow:            false,
+	}
+}
+
+const MaxFilterScopes = 64
+
+func isIDInRange(id int) bool {
+	return id >= 0 && id < MaxFilterScopes
+}
+
+func (fs FilterScope) IsEventTraceable(event events.ID) bool {
+	if _, found := fs.EventsToTrace[event]; found {
+		return true
+	}
+
+	return false
+}
+
+// TODO: add locking mechanism as scopes will change at runtime
+type FilterScopes struct {
+	filterScopesArray        [MaxFilterScopes]*FilterScope // underlying filter scopes array
+	filterScopesMap          map[*FilterScope]int          // stores only enabled scopes
+	filterScopesUserSpaceMap map[*FilterScope]int          // stores a reduced map with only user space filtered scopes
```

Comment:
Any advantage of setting the key-value like you did here over having a key which is the scope ID and a value which is *FilterScope?
Because I think it is more intuitive and can save a few lines of codes if we use scope ID as the key

Reply:
> I think this is basically using the map as a "set" to avoid duplication (a technique we use in other places too usually done with `map[T]bool/struct{}`).

Yes. This is the motivation.

> I'm not sure if we should actually care about duplicate scopes but I assume that's the reason, as there is an error about such a case.

Since we will be changing scopes at runtime, I assumed that by having pointers as keys we could make the management easier.